### PR TITLE
Send 500 internal server error on occuring exception

### PR DESCRIPTION
**Pull request by legionth**
An occuring exception should always lead to an internal server error.

### CODE REVIEW

File: src/HttpServer.php [hunk 1]
```diff
@@ -117,14 +117,25 @@ public function isChunkedEncodingActive(Request $request)
     public function handleRequest(ConnectionInterface $connection, Request $request)
     {
         $callback = $this->callback;
-        $response = $callback($request);
+
+        try {
+            $response = $callback($request);
+            if (!($response instanceof Response)) {
+                throw new InvalidResponseType('The value of the callback fucntion is not a valid Response object');
```
Comment (clue) [inline src/HttpServer.php]: typo :) Also, what is "value"?

File: tests/HttpServerTest.php [hunk 2]
```diff
@@ -59,4 +59,20 @@ public function testRequestWithChunkedEncodig()
...
         $this->connection->emit('data', array($request));
     }
+
+    public function testWrongResponseType()
+    {
+        $request = "GET /something HTTP/1.1\r\nHost: example.org\r\n\r\n";
+
+        $callback = function() {
+            return "This is an invalid type";
+        };
```
Comment (clue) [inline tests/HttpServerTest.php]: This doesn't test what the PR title implies. Is this intended?

**Comment by legionth** [comment]
Will split this PR into two PR. This PR handles occuring exception in the callback function.

File: src/HttpServer.php [hunk 3]
```diff
@@ -117,14 +117,22 @@ public function isChunkedEncodingActive(Request $request)
     public function handleRequest(ConnectionInterface $connection, Request $request)
     {
         $callback = $this->callback;
-        $response = $callback($request);
+
+        try {
+            $response = $callback($request);
+        } catch (\Exception $exception) {
+            $connection->write(RingCentral\Psr7\str(new Response(500)));
```
Comment (clue) [inline src/HttpServer.php:124]: I suppose an empty body is intended here? Note how many user agents (browsers) usually don't show HTTP headers etc.

**Review by clue — APPROVED** [review]
LGTM :+1:

**Pull request merged**